setup.py

```diff
@@ -19,9 +19,31 @@
 NAME = "argon2-cffi"
 PACKAGES = find_packages(where="src")
 
-# Optimized version requires SSE2 extensions.  They have been around since
-# 2001 so we try to compile it on every recent-ish x86.
-optimized = platform.machine() in ("i686", "x86", "x86_64", "AMD64")
+
+def has_sse_flag():
+    """Return a boolean indicating whether the sse flag name is supported on
+    the default compiler.
+    """
+    import tempfile
+    from distutils.ccompiler import new_compiler
```

Comment:
Any particular reason, why these imports are hidden away in here?

Reply:
Mostly because this codepath that needs those imports gets bypassed if the env variable is used.